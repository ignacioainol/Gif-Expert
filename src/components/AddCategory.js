import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(['']);

    const handleKeyPress = (e) => {
        setInputValue(e.target.value);

        console.log("handleInputChange llamado");
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (

        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleKeyPress}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};