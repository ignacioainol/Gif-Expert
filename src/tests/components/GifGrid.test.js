import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Unit Test in <GifGrid />', () => {
    test('It must to do match with snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const category = 'Another Night'
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('it should show items when images are loaded with useFetchGifs', () => {

    })


});
