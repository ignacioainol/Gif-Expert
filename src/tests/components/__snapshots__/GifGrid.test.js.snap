// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test in <GifGrid /> It must to do match with snapshot 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    Another Night
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
